### PR TITLE
DIV-6759: Adding new transformation when the respondent solicitor ref…

### PR DESCRIPTION
**Pull request by jvictorcf**
…erence data is passed in divorce session.

### Change description ###
No feature toggle is needed. This new field will only come from the front end when the front-end is feature is switched on.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/divorce/model/ccd/roles/UserRoles.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package uk.gov.hmcts.reform.divorce.model.ccd.roles;
+
+import lombok.NoArgsConstructor;
+
+import static lombok.AccessLevel.PRIVATE;
+
+@NoArgsConstructor(access = PRIVATE)
+public class UserRoles {
```
Comment (yarpo) [inline src/main/java/uk/gov/hmcts/reform/divorce/model/ccd/roles/UserRoles.java:8]: we have it in COS:

```
package uk.gov.hmcts.reform.divorce.orchestration.domain.model.ccd;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class CaseRoles {

    public static final String CREATOR = "[CREATOR]";
    public static final String PETITIONER_SOLICITOR = "[PETSOLICITOR]";

}
```

why dopn't you just use it here and then we can remove it from COS :)

**Review by yarpo — APPROVED** [review]

**Review by madjava — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/divorce/model/ccd/roles/UserRoles.java (hunk 1 above)
Comment (jvictorcf) [reply src/main/java/uk/gov/hmcts/reform/divorce/model/ccd/roles/UserRoles.java:8]: Good call. Will do that.

File: src/main/java/uk/gov/hmcts/reform/divorce/mapper/DivorceCaseToCCDMapper.java [hunk 2]
```diff
@@ -728,6 +730,18 @@ protected void mapLanguagePreferenceWelsh(DivorceSession divorceSession,
             toYesNoUpperCase(divorceSession.getLanguagePreferenceWelsh()));
     }
 
+    @AfterMapping
+    protected void mapRespondentOrganisationPolicy(DivorceSession divorceSession, @MappingTarget CoreCaseData result) {
+        if (divorceSession.getRespondentSolicitorReferenceDataId() != null) {
```
[the remark this answers is not in the document]
Comment (jvictorcf) [reply src/main/java/uk/gov/hmcts/reform/divorce/mapper/DivorceCaseToCCDMapper.java:735]: The feature toggle is in the front-end.

File: src/main/java/uk/gov/hmcts/reform/divorce/model/ccd/roles/UserRoles.java (hunk 1 above)
Comment (jvictorcf) [reply src/main/java/uk/gov/hmcts/reform/divorce/model/ccd/roles/UserRoles.java]: Done

File: src/main/java/uk/gov/hmcts/reform/divorce/mapper/DivorceCaseToCCDMapper.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jvictorcf) [reply src/main/java/uk/gov/hmcts/reform/divorce/mapper/DivorceCaseToCCDMapper.java:735]: Hope this is clear to you guys @madjava, @Dizrupto and @Jamesnelson117

File: src/main/java/uk/gov/hmcts/reform/divorce/model/ccd/roles/CaseRoles.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class CaseRoles {
+
+    public static final String CREATOR = "[CREATOR]";
+    public static final String PETITIONER_SOLICITOR = "[PETSOLICITOR]";
+    public static final String RESP_SOLICITOR = "[RESPSOLICITOR]";
```
Comment (yarpo) [inline src/main/java/uk/gov/hmcts/reform/divorce/model/ccd/roles/CaseRoles.java:11]: having known we have had `PETITIONER_SOLICITOR`, I'd say this one should be `RESPONDENT_SOLICITOR`

File: src/main/java/uk/gov/hmcts/reform/divorce/mapper/DivorceCaseToCCDMapper.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (yarpo) [reply src/main/java/uk/gov/hmcts/reform/divorce/mapper/DivorceCaseToCCDMapper.java:735]: ok

File: src/main/java/uk/gov/hmcts/reform/divorce/model/ccd/OrganisationPolicy.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+@ToString
+@EqualsAndHashCode
+@NoArgsConstructor
+public class OrganisationPolicy {
+
+    public OrganisationPolicy(String organisationId, String organisationName, String organisationPolicyReference, String assignedRole) {
```
Comment (yarpo) [inline src/main/java/uk/gov/hmcts/reform/divorce/model/ccd/OrganisationPolicy.java:16]: why not builder?

File: src/main/java/uk/gov/hmcts/reform/divorce/mapper/DivorceCaseToCCDMapper.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (madjava) [reply src/main/java/uk/gov/hmcts/reform/divorce/mapper/DivorceCaseToCCDMapper.java]: @jvictorcf, Yes if we cant search an organisation or the user opts to enter the data manually, this field `respondentSolicitorReferenceDataId` would be sent to you as null

[the remark this answers is not in the document]
Comment (jvictorcf) [reply src/main/java/uk/gov/hmcts/reform/divorce/mapper/DivorceCaseToCCDMapper.java]: @madjava, I'd even suggest not sending it at all in those situations. But I'll leave it to you.

File: src/main/java/uk/gov/hmcts/reform/divorce/model/ccd/OrganisationPolicy.java (hunk 4 above)
Comment (jvictorcf) [reply src/main/java/uk/gov/hmcts/reform/divorce/model/ccd/OrganisationPolicy.java:16]: There's more intelligence in this constructor than I'd get for a lombok builder.

File: src/main/java/uk/gov/hmcts/reform/divorce/model/ccd/roles/CaseRoles.java (hunk 3 above)
Comment (jvictorcf) [reply src/main/java/uk/gov/hmcts/reform/divorce/model/ccd/roles/CaseRoles.java]: done

**Pull request merged**